web-common/src/features/dashboards/workspace/DashboardContainer.svelte

```diff
@@ -1,64 +1,72 @@
 <script lang="ts">
   import { createResizeListenerActionFactory } from "@rilldata/web-common/lib/actions/create-resize-listener-factory";
   import { getContext } from "svelte";
+  import { getEltSize } from "@rilldata/web-common/features/dashboards/get-element-size";
   import type { Tweened } from "svelte/motion";
 
   export let exploreContainerWidth;
 
   export let leftMargin: string = undefined;
 
-  const navigationVisibilityTween = getContext(
+  // the navigationPaddingTween is a tweened value that is used
+  // to animate the extra padding that needs to be added to the
+  // dashboard container when the navigation pane is collapsed
+  const navigationPaddingTween = getContext(
     "rill:app:navigation-visibility-tween"
   ) as Tweened<number>;
 
-  const { observedNode, listenToNodeResize } =
-    createResizeListenerActionFactory();
+  const {
+    observedNode: dashboardContainerNode,
+    listenToNodeResize: dashboardContainerNodeWatcher,
+  } = createResizeListenerActionFactory();
 
-  $: exploreContainerWidth = $observedNode?.getBoundingClientRect()?.width || 0;
+  $: exploreContainerWidth = getEltSize($dashboardContainerNode, "x");
 
   $: leftSide = leftMargin
     ? leftMargin
-    : `calc(${$navigationVisibilityTween * 24}px + 1.25rem)`;
+    : `calc(${$navigationPaddingTween * 24}px + 1.25rem)`;
 </script>
 
-<section use:listenToNodeResize class="grid items-stretch surface">
+<section use:dashboardContainerNodeWatcher class="flex flex-col gap-y-1">
   <div
     class="explore-header border-b mb-3"
     style:padding-left={leftSide}
-    style:width={exploreContainerWidth + "px"}
+    style:width={"100%"}
```

Comment:
was able to eliminate the need to set this with JS just by setting it to be full width of the container